metals/src/main/scala/scala/meta/internal/metals/UserConfiguration.scala

```diff
@@ -133,11 +133,16 @@ object UserConfiguration {
     val base: JsonObject =
       Option(json.getAsJsonObject("metals")).getOrElse(new JsonObject)
 
-    def getKey[A](key: String, f: JsonElement => Option[A]): Option[A] = {
+    def getKey[A](
+        key: String,
+        f: JsonElement => Option[A],
+        isEmpty: JsonElement => Boolean = _ => false
```

Comment:
Instead of passing in `isEmpty = ...`, the `f` method can implement the filter instead.

Reply:
Hmm... that will not work since the f method is invoked after we get the value.